util/verible-format.py

```diff
@@ -0,0 +1,196 @@
+
+
+def get_verible_version():
+    verible_version = subprocess.Popen('verible-verilog-format --version',
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE, shell=True).stdout.read()
+    return verible_version.decode().split('\n')[0]
+
+
+def process_file(filename_abs, filename, inplace=False,
+                 show_diff=False, show_cst=False):
+    args = [get_verible_executable_path()] + VERIBLE_ARGS
+    if inplace:
+        args.append('--inplace')
+    args.append(filename_abs)
+
+    verible = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = verible.communicate()
+
+    ret = 0
+
+    if len(stderr) > 0:
+        logger.info('ERROR: ' + str(filename))
+        logger.debug(stderr.decode())
+        ret = 1
+    elif not inplace:
+        with open(filename_abs, 'r') as fp:
+            orig = fp.read().split('\n')
+            formatted = stdout.decode().split('\n')
+
+            diff = []
+            for line in difflib.unified_diff(orig, formatted, lineterm='', n=3,
+                                             fromfile=filename,
+                                             tofile=filename + '.formatted'):
+                diff.append(line)
+
+            ret = len(diff) > 0
+
+            if not show_diff and len(diff) > 0:
+                logger.info('File differs: ' + str(filename))
+
+            if show_diff and len(diff) > 0:
+                logger.info('')
+                for line in diff:
+                    logger.info(line)
+
+            if len(diff) > 0 and show_cst:
+                cst = subprocess.Popen(['verible-verilog-syntax',
+                                        '--printtree', filename],
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+                stdout, stderr = cst.communicate()
+                logger.info(stdout.decode())
+
+    return ret
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Format source code with Verible formatter')
+    parser.add_argument('-q', '--quiet', action='store_true', default=False,
+                        help='print only errors and warnings')
+    parser.add_argument('-v', '--verbose', action='store_true', default=False,
+                        help='print extra debug messages')
+    parser.add_argument('--show-diff', action='store_true', default=False,
+                        help='print diff (when files differs)')
+    parser.add_argument('--show-cst', action='store_true', default=False,
+                        help='print CST tree')
+    parser.add_argument('--progress', action='store_true', default=False,
+                        help='show progress')
+    parser.add_argument('--inplace', action='store_true', default=False,
+                        help='format files inplace (overwrite original files)')
```

Comment:
Nit: Probably "format files in place" (with a space between "in" and "place")

Reply:
Yes. Thanks!